veps/sig-compute/image-volume-proposal.md

```diff
@@ -0,0 +1,169 @@
+- Increases security by removing the need for virt-handler to bind mount disk files into the virt-launcher's compute container.
+- Simplifies virt-launcher pod manifest by removing unnecessary init containers (such as container-disk-binary, volumecontainerdisk-init,
+  volumekernel-boot-volume-init) and containers (volumecontainerdisk, volumekernel-boot-volume).
+- Lowers the dependency between virt-handler and virt-launcher, moving us closer to making the virt-launcher image standalone and able to
+  run outside of Kubernetes.
+
+## Goals
+
+- Remove all the current code that handles container disks and kernel boot with the existing workaround.
+- Allow users to test virtual machines with the new image volume feature by enabling a feature gate.
+- Create a safe upgrade path that supports old virtual machines and eventually moves them to use the 
+image volume feature without user intervention.
+- Make sure virt-launcher containers don’t get extra privileges, keeping the system secure.
+
+## Non Goals
+
+- This enhancement will not introduce changes to API fields or the API in general.
+- Users will not need to be aware of the underlying implementation details of container disks or kernel boot.
+- No additional functionality will be added to the existing API.
+- The current behavior, from users perspective, will remain unchanged.
+
+## Definition of Users
+
+- Anyone who creates virtual machines with container disk or kernel boot.
+- KubeVirt developers who need to develop features and fix bugs based on the current workaround.
+
+## User Stories
+
+### Faster Start-Up
+As a KubeVirt user, I want faster virtual machine start-ups, so that I can save time and improve system performance.
+
+### Easier Development for New Contributors
+As a new KubeVirt contributor, I want a simpler codebase, so that I can quickly understand how things work and contribute 
+to fixing bugs and adding features.
+
+### Better Security
+As a KubeVirt user, I want to remove privileged operations like bind mounts from virt-handler.
+
+## Repos
+
+https://github.com/kubevirt/kubevirt
+
+## Design
+
+This design introduces a new method for handling container disks and kernel boot. When a Virtual Machine with a container 
+disk or kernel boot is created, the virt-launcher pod will include an imageVolume in its specification, replacing the 
+current method that uses additional init containers and sidecar containers.
+
+Virt-handler will no longer be part of this process, and there will be no need to manage bind mounts or search for sockets. 
+From the launcher pod’s perspective, it will access the disk or kernel boot files directly from a predefined image path in 
+the volume mounts of virt-launcher.
+
+The design will ensure that the current method for container disks and kernel boot continues to work by default. 
+The new method using imageVolumes will only be used once the feature gate is enabled. By default, the feature 
+gate will be off until it is in Beta stage.
```

Comment:
nit: I think this paragraph can be removed, as what's written here is true for every enhancement

Reply:
Done